build_tools/teatime.py

```diff
@@ -40,6 +40,81 @@
 import subprocess
 import sys
 import time
+import threading
+
+
+def periodic_log_sync(
+    log_path: Path, s3_bucket: str, s3_subdir: str, interval: int = 30
+):
+    def sync():
+        while True:
+            time.sleep(interval)
+            if not log_path.exists():
+                continue
+            try:
+                subprocess.run(
+                    [
+                        "aws",
+                        "s3",
+                        "cp",
+                        str(log_path),
+                        f"s3://{s3_bucket}/{s3_subdir}/{log_path.name}",
+                        "--content-type",
+                        "text/plain",
+                    ],
+                    check=True,
+                )
+            except Exception as e:
+                print(f"[teatime] periodic upload failed: {e}", file=sys.stderr)
+
+    thread = threading.Thread(target=sync, daemon=True)
+    thread.start()
```

Comment:
What happens to this thread when the process exits? Does it get cleaned up? Are race conditions possible between this thread and the main thread copy commands?